docs/iris/src/sphinxext/generate_logo.py

```diff
@@ -0,0 +1,441 @@
+pad_inches = (figure_inches - globe_inches) / 2
+background_points = figure_inches * mpl_points_per_inch
+globe_points = globe_inches * mpl_points_per_inch
+
+mpl_font_points = rcParams["font.size"]
+inch_font_ratio = mpl_points_per_inch / mpl_font_points
+
+# XML ElementTree setup.
+namespaces = {"svg": "http://www.w3.org/2000/svg"}
+ET.register_namespace("", namespaces["svg"])
+
+################################################################################
+# Create new SVG elements for logo.
+
+# The elements that are displayed (not just referenced).
+# Order is important for layering.
+artwork_dict = OrderedDict.fromkeys(["background", "glow", "sea", "land",])
+# The elements that will just be referenced by artwork elements.
+defs_dict = {}
+
+# BACKGROUND
+artwork_dict["background"] = ET.Element(
+    "rect",
+    attrib={
+        "height": "100%",
+        "width": "100%",
+        "fill": "url(#background_gradient)",
+    },
+)
+background_gradient = ET.Element(
+    "linearGradient", attrib={"y1": "0%", "y2": "100%",},
+)
+background_gradient.append(
+    ET.Element("stop", attrib={"offset": "0", "stop-color": "#13385d",},)
+)
+background_gradient.append(
+    ET.Element("stop", attrib={"offset": "1", "stop-color": "#272b2c",},)
+)
+defs_dict["background_gradient"] = background_gradient
+
+# LAND
+# (Using Matplotlib and Cartopy).
+
+# Create land with simplified coastlines.
+simple_geometries = [
+    geometry.simplify(1.0, True) for geometry in LAND.geometries()
+]
+LAND.geometries = lambda: iter(simple_geometries)
+
+# Variable that will store the sequence of land-shaped SVG clips for each longitude.
+land_clips = OrderedDict()
+
+# Create a sequence of longitude values.
+central_longitude = -30
+central_latitude = 22.9
+rotation_frames = 180
+rotation_longitudes = linspace(start=central_longitude + 360,
+                               stop=central_longitude,
+                               num=rotation_frames,
+                               endpoint=False)
+# Normalise to -180..+180
+rotation_longitudes = (rotation_longitudes + 360.0 + 180.0) % 360.0 - 180.0
+
+for lon in rotation_longitudes:
+    # Use Matplotlib and Cartopy to generate land-shaped SVG clips for each longitude.
+
+    projection_rotated = ccrs.Orthographic(central_longitude=lon,
+                                           central_latitude=central_latitude)
+
+    fig = plt.figure(0, figsize=(figure_inches,) * 2)
+    ax = plt.axes(projection=projection_rotated)
+    # TODO: remove version check Iris is pinned appropriately.
+    if cartopy_version < "0.18":
+        fig.canvas.draw()
+    # Use constants set earlier to achieve desired dimensions.
+    plt.tight_layout(pad=pad_inches * inch_font_ratio)
+    ax.add_feature(LAND)
+
+    # Save as SVG and extract the resultant code.
+    svg_bytes = BytesIO()
+    plt.savefig(svg_bytes, format="svg")
+    svg_mpl = ET.fromstring(svg_bytes.getvalue())
+
+    # Find land paths and convert to clip paths.
```

Comment:
Can you not call `.set_clip_path` on the Features instead of postprocessing the SVG?

Reply:
If this were a one line thing I'd be in favour. But from what I can see, I would need to iterate through the `Polygon`s within `LAND.geometries()` and use each one as a clip path, as well as adding something artificial to be clipped? Given I would still need to extract the `Element`(s) I wanted for the eventual SVG, I don't think this change would reduce the complexity. Happy to be corrected 🙂

Reply:
I did not look too closely, and perhaps the display works in the reverse of what I expected. Originally, I was thinking that everything went through Matplotlib, but I guess those gradient backgrounds are pure SVG.

I think my suggestion still makes partial sense though, in that you could clip the continents by the petals, as those would seem to be much simpler shapes for clipping. The main aim to get rid of the artifacts at the transition from continent to whitespace between the petals, which partially shows the ocean colour.

Reply:
Yes, I had indeed hoped to do most/all the logo using Matplotlib, but the existing logo (which I am attempting to closely replicate) relies on bezier curves (the petals) and several gradients, neither of which Matplotlib writes to SVG. So in the end I've only used Cartopy/Matplotlib to provide the continents, and used relatively simple raw SVG objects for everything else.

What you've observed with the petal clipping is an artifact created by some SVG renderers - I've seen it in Edge/Chrome, but not Firefox/Inkscape/PowerPoint, and does not get bigger when zooming - the actual SVG code clips _everything_ with the petals and doesn't treat the ocean any differently to the continent.